cli: Improved autocomplete support for job dispatch and operator debug

davemay99 opened the pull request:
This PR provides additional autocomplete support for the following commands:

`nomad operator debug`
* -log-level
* -node-class
* -node-id
* -server-id
* -output

`nomad job dispatch`
* only autocomplete parameterized jobs

DerekStrickland commented on command/operator_debug.go:
```diff
@@ -195,6 +196,75 @@ func (c *OperatorDebugCommand) AutocompleteArgs() complete.Predictor {
+			return []string{}
+		}
+
+		unfiltered := make([]string, len(members.Members))
```
I probably would. Even if you are dealing with thousands of members, this isn't in the hot path I don't think, so the memory hit will be very temporary and only if dealing with lots of members.

davemay99 commented on command/operator_debug.go:
```diff
@@ -195,6 +196,75 @@ func (c *OperatorDebugCommand) AutocompleteArgs() complete.Predictor {
+		}
+
+		// Iterate over node classes looking for match
+		filtered := make([]string, len(classes))
```
Reverted to dynamic slice

davemay99 commented on command/job_dispatch.go:
```diff
@@ -75,11 +74,20 @@ func (c *JobDispatchCommand) AutocompleteArgs() complete.Predictor {
+		resp, _, err := client.Jobs().PrefixList(a.Last)
 		if err != nil {
-			return []string{}
+			return nil
```
No, but I changed this back as part of aligning style across predictors.

davemay99 commented:
@DerekStrickland I addressed the issues you raised, and rebased to align with changes recently merged into main.  Should be good to go, but please let me know if there are any remaining concerns.
